language-server/dm/executionManager.ml

```diff
@@ -51,13 +51,20 @@ type state = {
   initial : Vernacstate.t;
   of_sentence : (sentence_state * feedback_message list) SM.t;
   options : options;
+  doc_id : int; (* unique number used to interface with Coq's Feedback *)
 }
 
-let init vernac_state = {
-  initial = vernac_state;
-  of_sentence = SM.empty;
-  options = default_options;
-}
+let init =
+  let doc_id = ref (-1) in
```

Comment:
I found previous code (i.e. exposing a `fresh_xx_id` primitive) clearer than partial application with side effects, especially for OCaml non-experts.